terraform/_gen/main.go

```diff
@@ -73,6 +73,13 @@ type payload struct {
 	// IsPlainStruct states whether the resource type used by the API methods
 	// for this resource is a plain struct, rather than an interface.
 	IsPlainStruct bool
+	// ReadReturnValue indicates to read return value from Teleport Create/Update method.
+	// Reading return value is required in situations where data returned by Teleport is required
+	// to further proceed with Create/Update method in the provider.
+	ReadReturnValue bool
+	// MetadataNameAsID indicates to assign Metadata Value to an injected id field and vice versa.
+	// Currently used by device trust provider to assign Metadata.Name to injected id field.
```

Comment:
could you maybe explain, do we use metadata.name from the terraform resource as the id in the teleport resource, or the other way? which part injects the id where?